k256/src/arithmetic/scalar.rs

```diff
@@ -574,34 +588,58 @@ impl MulAssign<&Scalar> for Scalar {
 }
 
 impl Reduce<U256> for Scalar {
-    fn from_uint_reduced(w: U256) -> Self {
+    fn reduce(w: U256) -> Self {
         let (r, underflow) = w.sbb(&ORDER, Limb::ZERO);
-        let underflow = Choice::from((underflow.0 >> (Limb::BIT_SIZE - 1)) as u8);
+        let underflow = Choice::from((underflow.0 >> (Limb::BITS - 1)) as u8);
         Self(U256::conditional_select(&w, &r, !underflow))
     }
 }
 
 impl Reduce<U512> for Scalar {
-    fn from_uint_reduced(w: U512) -> Self {
+    fn reduce(w: U512) -> Self {
         WideScalar(w).reduce()
     }
 }
 
 impl ReduceNonZero<U256> for Scalar {
-    fn from_uint_reduced_nonzero(w: U256) -> Self {
+    fn reduce_nonzero(w: U256) -> Self {
         const ORDER_MINUS_ONE: U256 = ORDER.wrapping_sub(&U256::ONE);
         let (r, underflow) = w.sbb(&ORDER_MINUS_ONE, Limb::ZERO);
-        let underflow = Choice::from((underflow.0 >> (Limb::BIT_SIZE - 1)) as u8);
+        let underflow = Choice::from((underflow.0 >> (Limb::BITS - 1)) as u8);
         Self(U256::conditional_select(&w, &r, !underflow).wrapping_add(&U256::ONE))
     }
 }
 
 impl ReduceNonZero<U512> for Scalar {
-    fn from_uint_reduced_nonzero(w: U512) -> Self {
+    fn reduce_nonzero(w: U512) -> Self {
         WideScalar(w).reduce_nonzero()
     }
 }
 
+impl Sum for Scalar {
+    fn sum<I: Iterator<Item = Self>>(iter: I) -> Self {
+        iter.fold(Self::ZERO, |acc, w| acc + w)
```

Comment:
See the comment above about `fold` vs `reduce`

Reply:
Done